src/TranslationUi/Term/MetaboxAction.php

```diff
@@ -0,0 +1,68 @@
+<?php # -*- coding: utf-8 -*-
+declare(strict_types=1);
+
+namespace Inpsyde\MultilingualPress\YoastSeoSync\TranslationUi\Term;
+
+use Inpsyde\MultilingualPress\Framework\Admin\Metabox\Action;
+use Inpsyde\MultilingualPress\Framework\Admin\PersistentAdminNotices;
+use Inpsyde\MultilingualPress\Framework\Http\Request;
+use Inpsyde\MultilingualPress\TranslationUi\MetaboxFieldsHelper;
+use Inpsyde\MultilingualPress\TranslationUi\Term;
+
+final class MetaboxAction implements Action
+{
+    /**
+     * @var MetaboxFields
+     */
+    private $metaboxFields;
+
+    /**
+     * @var MetaboxFieldsHelper
+     */
+    private $fieldsHelper;
+
+    /**
+     * @var Term\RelationshipContext
+     */
+    private $relationshipContext;
+
+    public function __construct(
+        MetaboxFields $metaboxFields,
+        MetaboxFieldsHelper $fieldsHelper,
+        Term\RelationshipContext $relationshipContext
+    ) {
+
+        $this->metaboxFields = $metaboxFields;
+        $this->fieldsHelper = $fieldsHelper;
+        $this->relationshipContext = $relationshipContext;
+    }
+
+    public function save(Request $request, PersistentAdminNotices $notices): bool
+    {
+        $values = $request->bodyValue(
+            'multilingualpress',
+            INPUT_POST,
+            FILTER_UNSAFE_RAW,
+            FILTER_FORCE_ARRAY
+        );
+
+        $remoteSiteId = $this->relationshipContext->remoteSiteId();
+        $remoteTermId = $this->relationshipContext->remoteTermId();
+
+        $focuskw = $values["site-{$remoteSiteId}"][MetaboxFields::FIELD_FOCUS_KEYPHRASE];
+        $title = $values["site-{$remoteSiteId}"][Metaboxfields::FIELD_TITLE];
+        $metadesc = $values["site-{$remoteSiteId}"][Metaboxfields::FIELD_META_DESCRIPTION];
+        $canonical = $values["site-{$remoteSiteId}"][MetaboxFields::FIELD_CANONICAL];
+
+        $option = get_blog_option($remoteSiteId, 'wpseo_taxonomy_meta');
+        $term = get_term($remoteTermId);
+        $taxonomy = $term->taxonomy;
+
+        $option[$taxonomy][$remoteTermId][MetaboxFields::FIELD_FOCUS_KEYPHRASE] = $focuskw;
+        $option[$taxonomy][$remoteTermId][Metaboxfields::FIELD_TITLE] = $title;
+        $option[$taxonomy][$remoteTermId][Metaboxfields::FIELD_META_DESCRIPTION] = $metadesc;
+        $option[$taxonomy][$remoteTermId][MetaboxFields::FIELD_CANONICAL] = $canonical;
+
+        return update_blog_option($remoteSiteId, 'wpseo_taxonomy_meta', $option);
+    }
```

Comment:
I would move (if you agree to use the repository class mentioned above) this method into the Repository class and inject the instance of Repository here in MetaboxAction.

Reply:
This is still valid, what's your thoughts?

Reply:
I´m going to resolve this conversation, the above code is outdated, I already implemented the repository:
https://github.com/inpsyde/MultilingualPress-Yoast-SEO-Sync/blob/prevent-sync-yoast-metadata/src/TranslationUi/Term/Repository.php

It is used here for example: https://github.com/inpsyde/MultilingualPress-Yoast-SEO-Sync/blob/prevent-sync-yoast-metadata/src/TranslationUi/Term/Field/Title.php#L80